inception-imls-opennlp/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/opennlp/ner/OpenNlpNerRecommender.java

```diff
@@ -186,17 +189,87 @@ public EvaluationResult evaluate(List<CAS> aCasses, DataSplitter aDataSplitter)
         NameFinderME nameFinder = new NameFinderME(model);
 
         // Evaluate
-        try (NameSampleStream stream = new NameSampleStream(testSet)) {
-            TokenNameFinderEvaluator evaluator = new TokenNameFinderEvaluator(nameFinder);
-            evaluator.evaluate(stream);
-            // getFMeasure returns -1 if the evaluation cannot be performed, but we want any 
-            // recommender to get activated when the threshold is set to 0, so we cap at 0.
-            result.setDefaultScore(Math.max(0, evaluator.getFMeasure().getFMeasure()));
-            return result;
-        } catch (IOException e) {
-            LOG.error("Exception during evaluating the OpenNLP Named Entity Recognizer model.", e);
-            throw new RecommendationException("Error while evaluating OpenNlp NER", e);
+        List<AnnotatedTokenPair> predictions = new ArrayList<>();
+        for (NameSample sample : testSet) {
+            // clear adaptive data from feature generators if necessary
+            if (sample.isClearAdaptiveDataSet()) {
+                nameFinder.clearAdaptiveData();
+            }
+
+            // Span contains one NE, Array of them all in one sentence
+            String[] sentence = sample.getSentence();
+            Span[] predictedNames = nameFinder.find(sentence);
+            Span[] goldNames = sample.getNames();
+
+            predictions.addAll(determineLabelsForASentence(sentence, predictedNames,
+                    goldNames));
+
         }
+
+        return predictions.stream().collect(EvaluationResult
+                .collector(new HashSet<String>(asList(NO_NE_TAG)), trainingSetSize, testSetSize));
```

Comment:
`new HashSet<String>(asList(NO_NE_TAG))` -> `Collections.singleton(NO_NE_TAG)`

I would actually move the two numeric arguments to the front and use a variadic string argument to place `NO_NE_TAG`.